### ISSUE TITLE
Combination of xs:sequence and xs:choice

### ISSUE THREAD

**Issue by spilikin** [mvantellingen/python-zeep#760]
OASIS defines in https://docs.oasis-open.org/dss/v1.0/oasis-dss-core-schema-v1.0-os.xsd the following XML structure:

```
	<xs:element name="SignatureObject">
		<xs:complexType>
			<xs:sequence>
				<xs:choice>
					<xs:element ref="ds:Signature"/>
					<xs:element ref="dss:Timestamp"/>
					<xs:element ref="dss:Base64Signature"/>
					<xs:element ref="dss:SignaturePtr"/>
					<xs:element name="Other" type="dss:AnyType"/>
				</xs:choice>
			</xs:sequence>
			<xs:attribute name="SchemaRefs" type="xs:IDREFS" use="optional"/>
		</xs:complexType>
	</xs:element>
```

Web Service returns an element which seems to be the valid according to schema above:
```
      <ns3:SignatureObject>
        <ns3:Base64Signature Type="urn:ietf:rfc:3447"> ABCD</ns3:Base64Signature>
      </ns3:SignatureObject>
```

The following error occurs:
```
zeep.exceptions.XMLParseError: Unexpected element '{urn:oasis:names:tc:dss:1.0:core:schema}Base64Signature', expected '{urn:oasis:names:tc:dss:1.0:core:schema}Timestamp'
```

Zeep skips ds:Signature, cause it's from different namespace (see zeep/xsd/elements/element.py lines 148-151). Once the dss:Timestamp element is checked, zeep expects it as non optional (is_optional == True / min_occurs=1). But inside of Choice the element can be optional as long other element is present (quite tricky check idd).

This dirty fix works, but there ist probably a much better way in OrderIndicator?

```
zeep/xsd/elements/indicators.py:596
596            item_result = OrderedDict()
597            for elm_name, element in self.elements:
+++                if isinstance(self[0], Choice):
+++                    element.min_occurs = 0
```

The full WSDL can be found here: https://github.com/spilikin/telematik-interfaces

```
import pretend  # pip install pretend

from zeep import Client
from zeep.settings import Settings
from zeep.transports import Transport

# Replace YOUR-WSDL and OPERATION_NAME with the wsdl url
# and the method name you are calling. The response
# needs to be set in the content=""" """ var.

settings = Settings()
settings.forbid_entities = False
client = Client('../telematik-interfaces/r2.1.1/conn/AuthSignatureService.wsdl', settings=settings)
response = pretend.stub(
    status_code=200,
    headers={},
    content="""
<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
  <soap:Body>
    <ns4:ExternalAuthenticateResponse xmlns="http://ws.gematik.de/conn/ConnectorCommon/v5.0" xmlns:ns2="http://ws.gematik.de/conn/ConnectorContext/v2.0" xmlns:ns3="urn:oasis:names:tc:dss:1.0:core:schema" xmlns:ns4="http://ws.gematik.de/conn/SignatureService/v7.4" xmlns:ns5="http://www.w3.org/2000/09/xmldsig#" xmlns:ns6="http://ws.gematik.de/tel/error/v2.0" xmlns:ns7="http://ws.gematik.de/conn/CertificateServiceCommon/v2.0" xmlns:ns8="http://uri.etsi.org/02231/v2#" xmlns:ns9="urn:oasis:names:tc:SAML:1.0:assertion" xmlns:ns10="urn:oasis:names:tc:SAML:2.0:assertion" xmlns:ns11="http://www.w3.org/2001/04/xmlenc#" xmlns:ns12="http://uri.etsi.org/01903/v1.3.2#" xmlns:ns13="urn:oasis:names:tc:dss-x:1.0:profiles:verificationreport:schema#" xmlns:ns14="urn:oasis:names:tc:dss-x:1.0:profiles:SignaturePolicy:schema#">
      <Status>
        <Result>OK</Result>
      </Status>
      <ns3:SignatureObject>
        <ns3:Base64Signature Type="urn:ietf:rfc:3447">WGRtQklURFBFRENTT3QxdzVxYXEzUVN1emdVZlpLL0x1eW4vSDlKMWlpNHNiblQ4c3BQVWxMMUZyaGFsZkREN2Fid24zUnlzclZaM3BYemZtMWZrQU9NaXVDQUVxTnQxZEZCZXhpdzQ0V1dvUGZTTE1MdzZaZ28rNkxtOHNlTHRoZ3B5ZWliUklwcGMzKzdiT2hwSEVuN0UrM0srRDBEL3JSenh6YzJYN0k1Zy94UW4xTVJsL0VqclU1MEo4Q0FnSG9wUDVwZzBSTkJWaGFNa1FPM1dQTVhTZGJLSnZVOXY0R2lzMTNtV0dzaU1qM3JZMHBqNytJNm83b1Q2N3Y1RUUvSG9lNFBjNWt1QWI2Yll3WFBzUUFzMitjaDRKWm9tN29QVk51WUZGRTMraTJNcUZuUmRkL3VRVzVNMSszTGsrNml1WEhlRVVQN1gxUHlqNldoeStnPT0=</ns3:Base64Signature>
      </ns3:SignatureObject>
    </ns4:ExternalAuthenticateResponse>
  </soap:Body>
</soap:Envelope>
    """)


operation = client.service._binding._operations['ExternalAuthenticate']
result = client.service._binding.process_reply(
    client, operation, response)

print(result)
```

**Comment by spilikin** [comment]
Created Pull Request #761 with better patch

**Comment by mvantellingen** [comment]
Merged #761
